Framework/Algorithms/src/He3TubeEfficiency.cpp

```diff
@@ -89,54 +88,53 @@ void He3TubeEfficiency::init() {
 }
 
 /**
- * Executes the algorithm
- */
+* Executes the algorithm
+*/
 void He3TubeEfficiency::exec() {
   // Get the workspaces
-  this->inputWS = this->getProperty("InputWorkspace");
-  this->outputWS = this->getProperty("OutputWorkspace");
+  m_inputWS = this->getProperty("InputWorkspace");
+  m_outputWS = this->getProperty("OutputWorkspace");
 
-  if (this->outputWS != this->inputWS) {
-    this->outputWS = API::WorkspaceFactory::Instance().create(this->inputWS);
+  if (m_outputWS != m_inputWS) {
+    m_outputWS = API::WorkspaceFactory::Instance().create(m_inputWS);
   }
 
   // Get the detector parameters
-  this->paraMap = &(this->inputWS->instrumentParameters());
+  m_paraMap = &(m_inputWS->instrumentParameters());
 
   // Store some information about the instrument setup that will not change
-  this->samplePos = this->inputWS->getInstrument()->getSample()->getPos();
+  m_samplePos = m_inputWS->getInstrument()->getSample()->getPos();
 
   // Check if it is an event workspace
   DataObjects::EventWorkspace_const_sptr eventW =
-      boost::dynamic_pointer_cast<const DataObjects::EventWorkspace>(inputWS);
+      boost::dynamic_pointer_cast<const DataObjects::EventWorkspace>(m_inputWS);
   if (eventW != nullptr) {
     this->execEvent();
     return;
   }
 
-  std::size_t numHists = this->inputWS->getNumberHistograms();
-  this->progress = new API::Progress(this, 0.0, 1.0, numHists);
+  std::size_t numHists = m_inputWS->getNumberHistograms();
+  m_progress = new API::Progress(this, 0.0, 1.0, numHists);
 
-  PARALLEL_FOR2(inputWS, outputWS)
+  PARALLEL_FOR2(m_inputWS, m_outputWS)
   for (int i = 0; i < static_cast<int>(numHists); ++i) {
     PARALLEL_START_INTERUPT_REGION
 
-    this->outputWS->setX(i, this->inputWS->refX(i));
+    m_outputWS->setSharedX(i, m_inputWS->sharedX(i));
     try {
       this->correctForEfficiency(i);
     } catch (std::out_of_range &) {
       // if we don't have all the data there will be spectra we can't correct,
       // avoid leaving the workspace part corrected
-      Mantid::MantidVec &dud = this->outputWS->dataY(i);
-      std::transform(dud.begin(), dud.end(), dud.begin(),
-                     std::bind2nd(std::multiplies<double>(), 0));
+      auto &dud = m_outputWS->mutableY(i);
+      dud.assign(dud.size(), 0);
```

Comment:
With the changes @SimonHeybrock made in [this](https://github.com/mantidproject/mantid/pull/17147) PR, you will be able to do things like `m_outputWS->mutableY(i) = 0;`. Look out for when that gets merged.